server/sonar-ce/src/test/java/org/sonar/ce/CeDistributedInformationImplTest.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2017 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.ce;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import org.junit.Test;
+import org.sonar.ce.cluster.HazelcastClientWrapperImpl;
+import org.sonar.ce.taskprocessor.CeWorkerFactory;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.data.MapEntry.entry;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static org.sonar.process.cluster.ClusterObjectKeys.WORKER_UUIDS;
+
+public class CeDistributedInformationImplTest {
+  private String clientUUID1 = "1";
+  private String clientUUID2 = "2";
+  private String clientUUID3 = "3";
+  private Map workerMap = ImmutableMap.of(
+    clientUUID1, ImmutableSet.of("1", "2"),
+    clientUUID2, ImmutableSet.of("3"),
+    clientUUID3, ImmutableSet.of("4", "5", "6")
+  );
+
+  private HazelcastClientWrapperImpl hzClientWrapper = mock(HazelcastClientWrapperImpl.class);
+
+  @Test
+  public void getWorkerUUIDs_must_returns_all_the_workers_in_cluster() {
```

Comment:
I'd name this method `getWorkerUUIDs_returns_union_of_workers_uuids_of_local_and_cluster_worker_uuids`, which would lead me to test:
* empty local + empty cluster
* empty local + cluster
* local + empty cluster
* (and finally) local + cluster which contains local (occurs after call to `broadcastWorkerUUIDs`)